### PR TITLE
docs: Added documentation for GameLoop class

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

Added documentation for the GameLoop class.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `doc:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [x] I have updated/added relevant examples in `examples`.

## Breaking Change

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Review by spydon — APPROVED** [review]
LGTM, but I wonder if there was a specific reason to have the separate methods, since they weren't doing exactly the same thing.

File: packages/flame/lib/src/game/game_loop.dart [hunk 1]
```diff
@@ -1,48 +1,79 @@
...
+  /// do not have to stop the game loop before disposing of it.
   void dispose() {
     _ticker.dispose();
   }
 
-  void pause() {
-    _ticker.muted = true;
-    previous = Duration.zero;
-  }
+  @Deprecated('Use stop() instead')
```
Comment (erickzanardo) [inline packages/flame/lib/src/game/game_loop.dart:74]: Why refactor these?

Either way a change like this should be on its own PR

**Review by erickzanardo — CHANGES_REQUESTED** [review]
Docs LGTM, but this PR is addressing multiple things under the same PR, I don't understood the need to refactor pause and resume, but this should be treated on a separate PR.

**Comment by st-pasha** [comment]
> I don't understood the need to refactor pause and resume

As I was writing the documentation for each method in the class, I found that it's hard to describe what those methods do as compared to start and stop. I mean, sure they had their own quirks, but it's better to fix those quirks than document them.

File: packages/flame/lib/src/game/game_loop.dart (hunk 1 above)
Comment (st-pasha) [reply packages/flame/lib/src/game/game_loop.dart:74]: I'll split them out into another PR.

File: packages/flame/lib/src/game/game_loop.dart [hunk 2]
```diff
@@ -1,48 +1,82 @@
...
+  /// do not have to stop the game loop before disposing of it.
   void dispose() {
     _ticker.dispose();
   }
 
-  void pause() {
-    _ticker.muted = true;
-    previous = Duration.zero;
-  }
+  @Deprecated('Internal variable')
```
Comment (wolfenrain) [inline packages/flame/lib/src/game/game_loop.dart:74]: Why use `Deprecated`? There are specialized decorators for this like `Protected`.

Comment (st-pasha) [reply packages/flame/lib/src/game/game_loop.dart:74]: Because we don't use that variable anywhere. It was never intended to be public.
So, now that we make it private, the getter is provided in case anyone's code was reading that variable. The getter is marked deprecated, as a way to signal that this variable really shouldn't be used, and that the getter will be removed after a while.

**Review by erickzanardo — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]
